### ISSUE TITLE
git-pulls improvements

### ISSUE THREAD

**Issue by pfuntner** [pfuntner/toys#95]
While working on doc for `git-pulls`, I came up with ideas to improve it:

- Accept one or more directories and seek any directory anywhere in the tree that has a `.git` directory
- Get rid of some of the noisy operation, especially when there are actual new commits to pull and for `--dry-run`
- In `--dry-run` mode: sense when the remote server has new commits and report on them in the regular fashion, just without doing pulls

**Comment by pfuntner** [comment]
I learned more about sensing new commits on the remote server.  Typically, you would have to do a `git fetch` to receive the latest information from the server because the `git` CLI for the most part only works with the local repo.  But the REST API could be used on **public** repos.

I did the following:
1. Created a new private repo with a README
2. Cloned the repo on my laptop:
```
$ git-log
Commit                                    Author              Date (UTC)           Text
82a0ea987634e5a7e17f287e44504184a96a2fda  pfuntner@pobox.com  2022-06-06 11:19:31  Create README.md
```
3. Updated the README
4. Tried to use the REST API but had trouble because it was private
5. I learned I could not use my ssh key.  I didn't want to use my password so I decided to make the repo public (most of my personal repos are public anyway):
```
$ curl https://api.github.com/repos/pfuntner/temp/commits | json --flat | fitwidth
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  8531  100  8531    0     0   106k      0 --:--:-- --:--:-- --:--:--  108k
/0/author/avatar_url 'https://avatars.githubusercontent.com/u/23261559?v=4'
/0/author/events_url 'https://api.github.com/users/pfuntner/events{/privacy}'
/0/author/followers_url 'https://api.github.com/users/pfuntner/followers'
/0/author/following_url 'https://api.github.com/users/pfuntner/following{/other_user}'
/0/author/gists_url 'https://api.github.com/users/pfuntner/gists{/gist_id}'
/0/author/gravatar_id ''
/0/author/html_url 'https://github.com/pfuntner'
/0/author/id 23261559
/0/author/login 'pfuntner'
/0/author/node_id 'MDQ6VXNlcjIzMjYxNTU5'
/0/author/organizations_url 'https://api.github.com/users/pfuntner/orgs'
/0/author/received_events_url 'https://api.github.com/users/pfuntner/received_events'
/0/author/repos_url 'https://api.github.com/users/pfuntner/repos'
/0/author/site_admin False
/0/author/starred_url 'https://api.github.com/users/pfuntner/starred{/owner}{/repo}'
/0/author/subscriptions_url 'https://api.github.com/users/pfuntner/subscriptions'
/0/author/type 'User'
/0/author/url 'https://api.github.com/users/pfuntner'
/0/comments_url 'https://api.github.com/repos/pfuntner/temp/commits/16f739e4a3a23473c2b737edcebbba94dfec6cba/comments'
/0/commit/author/date '2022-06-06T11:21:40Z'
/0/commit/author/email 'pfuntner@pobox.com'
/0/commit/author/name 'John Pfuntner'
/0/commit/comment_count 0
/0/commit/committer/date '2022-06-06T11:21:40Z'
/0/commit/committer/email 'noreply@github.com'
/0/commit/committer/name 'GitHub'
/0/commit/message 'Update README.md'
/0/commit/tree/sha 'c99f46b56535ab5ad7119d9ffdb1b60c00d9ef71'
/0/commit/tree/url 'https://api.github.com/repos/pfuntner/temp/git/trees/c99f46b56535ab5ad7119d9ffdb1b60c00d9ef71'
/0/commit/url 'https://api.github.com/repos/pfuntner/temp/git/commits/16f739e4a3a23473c2b737edcebbba94dfec6cba'
/0/commit/verification/payload 'tree c99f46b56535ab5ad7119d9ffdb1b60c00d9ef71\nparent 82a0ea987634e5a7e17f287e44504184a96a2fda\nauthor John Pfuntner <pfuntner@pobox.com> 1654514500 -0400\ncommitter GitHub <noreply@github.com> 1654514500 -0400\n\nUpdate README.md'
/0/commit/verification/reason 'valid'
/0/commit/verification/signature '-----BEGIN PGP SIGNATURE-----\n\nwsBcBAABCAAQBQJineNECRBK7hj4Ov3rIwAAJHcIAI0lzp8ZgAboOE7VcYQoeMqB\nIWIZvpMIVAAkMWRfZ8WqfCKmc6JV7V0T4UuLFL/k8x4Aj5KtLmVHntdqogO2J5Aj\nSb1r2j8GBHBL7F2x/JzeK8BPVeRnTp0mqVmxwGojqKzqcJooDKrm9AwpshN0f/Yo\nF0N7iqvwH45uiM2OjdfVoX3avYUlMEENbOYcOoHrRR1lYHg+NuktuPx1TKIZQnt/\nzBxPdUhuURo5UuDxc/ro9mSmCdq/kwEP3LkD1xk7sRvn+h9K36dUmbp9Buo3eQH2\nLCDKjdlvJj1XH4V+Nk+tPTdIfblRBguJC5CY842A1jURe0hTYH6wXsjYFiucqM8=\n=oXUY\n-----END PGP SIGNATURE-----\n'
/0/commit/verification/verified True
/0/committer/avatar_url 'https://avatars.githubusercontent.com/u/19864447?v=4'
/0/committer/events_url 'https://api.github.com/users/web-flow/events{/privacy}'
/0/committer/followers_url 'https://api.github.com/users/web-flow/followers'
/0/committer/following_url 'https://api.github.com/users/web-flow/following{/other_user}'
/0/committer/gists_url 'https://api.github.com/users/web-flow/gists{/gist_id}'
/0/committer/gravatar_id ''
/0/committer/html_url 'https://github.com/web-flow'
/0/committer/id 19864447
/0/committer/login 'web-flow'
/0/committer/node_id 'MDQ6VXNlcjE5ODY0NDQ3'
/0/committer/organizations_url 'https://api.github.com/users/web-flow/orgs'
/0/committer/received_events_url 'https://api.github.com/users/web-flow/received_events'
/0/committer/repos_url 'https://api.github.com/users/web-flow/repos'
/0/committer/site_admin False
/0/committer/starred_url 'https://api.github.com/users/web-flow/starred{/owner}{/repo}'
/0/committer/subscriptions_url 'https://api.github.com/users/web-flow/subscriptions'
/0/committer/type 'User'
/0/committer/url 'https://api.github.com/users/web-flow'
/0/html_url 'https://github.com/pfuntner/temp/commit/16f739e4a3a23473c2b737edcebbba94dfec6cba'
/0/node_id 'C_kwDOHdOJ_NoAKDE2ZjczOWU0YTNhMjM0NzNjMmI3MzdlZGNlYmJiYTk0ZGZlYzZjYmE'
/0/parents/0/html_url 'https://github.com/pfuntner/temp/commit/82a0ea987634e5a7e17f287e44504184a96a2fda'
/0/parents/0/sha '82a0ea987634e5a7e17f287e44504184a96a2fda'
/0/parents/0/url 'https://api.github.com/repos/pfuntner/temp/commits/82a0ea987634e5a7e17f287e44504184a96a2fda'
/0/sha '16f739e4a3a23473c2b737edcebbba94dfec6cba'
/0/url 'https://api.github.com/repos/pfuntner/temp/commits/16f739e4a3a23473c2b737edcebbba94dfec6cba'
/1/author/avatar_url 'https://avatars.githubusercontent.com/u/23261559?v=4'
/1/author/events_url 'https://api.github.com/users/pfuntner/events{/privacy}'
/1/author/followers_url 'https://api.github.com/users/pfuntner/followers'
/1/author/following_url 'https://api.github.com/users/pfuntner/following{/other_user}'
/1/author/gists_url 'https://api.github.com/users/pfuntner/gists{/gist_id}'
/1/author/gravatar_id ''
/1/author/html_url 'https://github.com/pfuntner'
/1/author/id 23261559
/1/author/login 'pfuntner'
/1/author/node_id 'MDQ6VXNlcjIzMjYxNTU5'
/1/author/organizations_url 'https://api.github.com/users/pfuntner/orgs'
/1/author/received_events_url 'https://api.github.com/users/pfuntner/received_events'
/1/author/repos_url 'https://api.github.com/users/pfuntner/repos'
/1/author/site_admin False
/1/author/starred_url 'https://api.github.com/users/pfuntner/starred{/owner}{/repo}'
/1/author/subscriptions_url 'https://api.github.com/users/pfuntner/subscriptions'
/1/author/type 'User'
/1/author/url 'https://api.github.com/users/pfuntner'
/1/comments_url 'https://api.github.com/repos/pfuntner/temp/commits/82a0ea987634e5a7e17f287e44504184a96a2fda/comments'
/1/commit/author/date '2022-06-06T11:19:31Z'
/1/commit/author/email 'pfuntner@pobox.com'
/1/commit/author/name 'John Pfuntner'
/1/commit/comment_count 0
/1/commit/committer/date '2022-06-06T11:19:31Z'
/1/commit/committer/email 'noreply@github.com'
/1/commit/committer/name 'GitHub'
/1/commit/message 'Create README.md'
/1/commit/tree/sha 'c994024c874e1deba6f4a65915ad80b0337e8fdd'
/1/commit/tree/url 'https://api.github.com/repos/pfuntner/temp/git/trees/c994024c874e1deba6f4a65915ad80b0337e8fdd'
/1/commit/url 'https://api.github.com/repos/pfuntner/temp/git/commits/82a0ea987634e5a7e17f287e44504184a96a2fda'
/1/commit/verification/payload 'tree c994024c874e1deba6f4a65915ad80b0337e8fdd\nauthor John Pfuntner <pfuntner@pobox.com> 1654514371 -0400\ncommitter GitHub <noreply@github.com> 1654514371 -0400\n\nCreate README.md'
/1/commit/verification/reason 'valid'
/1/commit/verification/signature '-----BEGIN PGP SIGNATURE-----\n\nwsBcBAABCAAQBQJineLDCRBK7hj4Ov3rIwAA0kcIAB1zOF9mGZDBbLsnMy1hEvqn\nxsFFcsc/dZFtZ+1QAUJKhJttAruu7C76DjbfHmn9S3JKbgP3gyyI00eVZeOGAQCW\na4GICN0VcSo+FH5Je5Hjzohyd+da8UWET35Y6G+bUfFYOddlCKJxhHt1CZBYNMuI\neCeLoE3CVMDkLVTzQQv8QFup82oorO1F/cOBF63WDys0L+/33xs0NIRCKuQEVWcV\nnBNbwgJ1L5lwXIyD0/6Uy57MPrGrL0G54asS4854T3MC0VK/4rZuNooMTokb9xzW\nUOtwGqhTEUUMmaK1ljh31zRBIMMXNLVJLtXg0gAN3P2H+xfBNAJIcN1U9k5G3EA=\n=VuNl\n-----END PGP SIGNATURE-----\n'
/1/commit/verification/verified True
/1/committer/avatar_url 'https://avatars.githubusercontent.com/u/19864447?v=4'
/1/committer/events_url 'https://api.github.com/users/web-flow/events{/privacy}'
/1/committer/followers_url 'https://api.github.com/users/web-flow/followers'
/1/committer/following_url 'https://api.github.com/users/web-flow/following{/other_user}'
/1/committer/gists_url 'https://api.github.com/users/web-flow/gists{/gist_id}'
/1/committer/gravatar_id ''
/1/committer/html_url 'https://github.com/web-flow'
/1/committer/id 19864447
/1/committer/login 'web-flow'
/1/committer/node_id 'MDQ6VXNlcjE5ODY0NDQ3'
/1/committer/organizations_url 'https://api.github.com/users/web-flow/orgs'
/1/committer/received_events_url 'https://api.github.com/users/web-flow/received_events'
/1/committer/repos_url 'https://api.github.com/users/web-flow/repos'
/1/committer/site_admin False
/1/committer/starred_url 'https://api.github.com/users/web-flow/starred{/owner}{/repo}'
/1/committer/subscriptions_url 'https://api.github.com/users/web-flow/subscriptions'
/1/committer/type 'User'
/1/committer/url 'https://api.github.com/users/web-flow'
/1/html_url 'https://github.com/pfuntner/temp/commit/82a0ea987634e5a7e17f287e44504184a96a2fda'
/1/node_id 'C_kwDOHdOJ_NoAKDgyYTBlYTk4NzYzNGU1YTdlMTdmMjg3ZTQ0NTA0MTg0YTk2YTJmZGE'
/1/sha '82a0ea987634e5a7e17f287e44504184a96a2fda'
/1/url 'https://api.github.com/repos/pfuntner/temp/commits/82a0ea987634e5a7e17f287e44504184a96a2fda'
$
```

An ssh key cannot be used with the REST API.

**Comment by pfuntner** [comment]
I addressed some of the changes I envisioned in 07ce219

**Comment by pfuntner** [comment]
006368f addresses the improvement of processing the directory argument.
